static/app/views/insights/agentMonitoring/views/onboarding.tsx

```diff
@@ -0,0 +1,505 @@
+
+function useOnboardingProject() {
+  const {projects} = useProjects();
+  const pageFilters = usePageFilters();
+  const selectedProject = getSelectedProjectList(
+    pageFilters.selection.projects,
+    projects
+  );
+  const agentMonitoringProjects = selectedProject.filter(p =>
+    agentMonitoringPlatforms.has(p.platform as PlatformKey)
+  );
+
+  if (agentMonitoringProjects.length > 0) {
+    return agentMonitoringProjects[0];
+  }
+  return selectedProject[0];
+}
+
+function useAgentEventWaiter(project: Project) {
+  const [refetchKey, setRefetchKey] = useState(0);
+
+  const {data, isPending} = useEAPSpans(
+    {
+      search: 'span.op:"gen_ai.*"',
+      fields: ['id'],
+      limit: 1,
+      enabled: !!project,
+      useQueryOptions: {
+        additonalQueryKey: [`refetch-${refetchKey}`],
+      },
+    },
+    Referrer.ONBOARDING
+  );
+
+  const hasEvents = Boolean(data?.length);
+
+  // Create a custom key that changes every 5 seconds to trigger refetch
+  // TODO(aknaus): remove this and add refetchInterval to useEAPSpans
+  useEffect(() => {
+    if (!hasEvents) {
+      const interval = setInterval(() => {
+        setRefetchKey(prev => prev + 1);
+      }, 5000); // Poll every 5 seconds
+
+      return () => clearInterval(interval);
+    }
+    return () => {};
+  }, [hasEvents]);
+
+  return {
+    hasEvents: Boolean(data?.length),
+    isPending,
+  };
+}
+
+function WaitingIndicator({project}: {project: Project}) {
+  const {hasEvents} = useAgentEventWaiter(project);
+  const {reloadProjects, fetching} = useProjects();
+
+  return hasEvents ? (
+    <Button
+      priority="primary"
+      busy={fetching}
+      onClick={() => {
+        reloadProjects();
+      }}
+    >
+      {t('View Agent Monitoring')}
+    </Button>
+  ) : (
+    <EventWaitingIndicator />
+  );
+}
+
+function ConfigurationRenderer({configuration}: {configuration: Configuration}) {
+  const subConfigurations = configuration.configurations ?? [];
+  return (
+    <ConfigurationWrapper>
+      {configuration.description && (
+        <DescriptionWrapper>{configuration.description}</DescriptionWrapper>
+      )}
+      {configuration.code ? (
+        Array.isArray(configuration.code) ? (
+          <TabbedCodeSnippet tabs={configuration.code} />
+        ) : (
+          <OnboardingCodeSnippet language={configuration.language}>
+            {configuration.code}
+          </OnboardingCodeSnippet>
+        )
+      ) : null}
+      {subConfigurations.map((subConfiguration, index) => (
+        <ConfigurationRenderer key={index} configuration={subConfiguration} />
+      ))}
+      {configuration.additionalInfo && (
+        <AdditionalInfo>{configuration.additionalInfo}</AdditionalInfo>
+      )}
+    </ConfigurationWrapper>
+  );
+}
+
+function StepRenderer({
+  project,
+  step,
+  isLastStep,
+}: {
+  isLastStep: boolean;
+  project: Project;
+  step: StepProps;
+}) {
+  const {type, title} = step;
```

Comment:
Nit, but destructuring for tersenes is imo very low value and could just be inlined into `step.title` and `step.type`